### ISSUE TITLE
Crashes when attaching as admin

### ISSUE THREAD

**Issue by Argagaes** [appsec-labs/Advanced_Packet_Editor#1]
The program always stops responding if I try to attach it as admin. I'm running win10, the same issue exists for my friend who uses win7. Any idea what might cause this? Maybe it's the .dat file this program injects? How can you open the .dat anyways :P?

**Comment by giladof** [comment]
The current implementation of the .dat file is a 32bit file, thus you can only inject the APE into a 32 bit process.

**Comment by Argagaes** [comment]
Yes I understand that. The issue exists even when injecting into 32-bit processes. Can 64-bit processes even show up in the process list when you run it as admin? Doesn't it just throw a access error when it tries to fetch the path and then just doesn't add the row?
Here is a video:
https://youtu.be/zetMxnUxHuA

This happens with ANY process when ran as admin. There is simply no way to attach if it's running on admin rights, no matter which version of this tool I've tried (The original source from 2010, this 1.5 or your site's 2.0). It also happened to my friend who is on Win7 rather than Win10 like me, so it's unlikely to be a fault in my pc.
(Obviously it worked fine non-admin mode for the process in the video, but some processes don't show up without admin rights which makes this an issue)

**Comment by giladof** [comment]
Well, it requires further investigation.
Maybe the problem is with the between users' access (injecting DLL by an application that runs under one user, the admin, into another application that runs on another user). I think that this may be the case because some features don't work in a similar scenario (such as using drag and drop from an application that runs as admin to windows explorer).

Another reason could be the ".dat" file.

A way to test it is to verify whether the DLL was injected or not and see from there.
I'll try to test is when I can.

**Comment by Argagaes** [comment]
Awesome :) "injecting DLL by an application that runs under one user, the admin, into another application that runs on another user" wouldn't it then work on processes you also ran as admin though?
Well, needs testing so doesn't matter! Thanks for the fast replies! Amazing to see such an old project still get this good support ^^ Especially since this is an amazing tool.

**Comment by giladof** [comment]
Thank you 👍 We really appreciate it! 
You are more than welcome to check out more of our tools @ https://appsec-labs.com/tools/

**Comment by Argagaes** [comment]
Did you have time to get around testing/fixing it yet?

**Comment by giladof** [comment]
Not at this time since I'm packed with work
Next week I'll have some time

Gilad Ofir
Senior Application Security Consultant, AppSec Labs
+972528321252

**Comment by Argagaes** [comment]
Had time for it now :D? Sorry if I'm bothering!

**Comment by giladof** [comment]
Hi
Unfortunately no...
I'll let you know when I get into it
Sorry for the wait... we're busy with other matters

Gilad Ofir
Senior Application Security Consultant, AppSec Labs
+972528321252

**Comment by Argagaes** [comment]
Uh it's been a month now...

**Comment by giladof** [comment]
Haven't been able to fix it yet.. sorry
Working on it

Gilad Ofir
Senior Application Security Consultant, AppSec Labs
+972528321252

**Comment by Argagaes** [comment]
been almost 2 months since last message ...

**Comment by Argagaes** [comment]
Oh, I think the .dll is encrypted. I tried looking about it and found in the original packet editors changelog something about it. I guess it's impossible to fix this issue unless making a whole new .dll file / getting in touch with the original maker.

**Comment by socallie** [comment]
I am the original developer. I might be interested in collaborating with you on a newer 32-bit and 64-bit version if you're interested.

**Comment by appseclabs** [comment]
+ Gilad



Regards,

Yaron Hakon, Application Security Expert, AppSec Labs CEO

Mobile: +972-50-6668229 | Office: +972-52-3413555

Email: yaron@appsec-labs.com | Website: https://appsec-labs.com



*The mind is like any other muscle in your body. Use it or lose it** !*



*From:* Steve Warner <notifications@github.com>
*Sent:* Saturday, 19 October 2019 3:51
*To:* appsec-labs/Advanced_Packet_Editor <
Advanced_Packet_Editor@noreply.github.com>
*Cc:* Subscribed <subscribed@noreply.github.com>
*Subject:* Re: [appsec-labs/Advanced_Packet_Editor] Crashes when attaching
as admin (#1)



I am the original developer. I might be interested in collaborating with
you on a newer 32-bit and 64-bit version if you're interested.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/appsec-labs/Advanced_Packet_Editor/issues/1?email_source=notifications&email_token=AB73XXPLJKSWH5E7NIIG2A3QPJKX3A5CNFSM4DLXGUIKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBWTWWA#issuecomment-544029528>,
or unsubscribe

**Comment by appseclabs** [comment]
Thanks, but I'm not maintaining the tool anymore

**Comment by r0ach0** [comment]
> I am the original developer. I might be interested in collaborating with you on a newer 32-bit and 64-bit version if you're interested.

Hi, original developer.
I'm trying to use your library(WSPE.dat) but the pipe doesn't connec to library for unknown reason.
Can you help me?

**Comment by wheet64** [comment]
Something must be preventing the load/decryption of WSPE.dat in the target's memory. Are you sure the target is 32-bit? DEP (Data Execution Prevention) or your virus software may cause issues as well.

**Comment by ksorah1** [comment]
me it crashes immediately when I attach the process...

**Comment by Kevin870517** [comment]
I have the same problem, when I Attached, it crashed.
